### DISCUSSION TITLE
Preview

### DISCUSSION THREAD

**Discussion by Finnein** [keiwando/toonsquid#303] [Ideas]
Would be nice if there was a preview and not only playback button that shows only what’s on the canvas and not what’s outside of it.

**Comment by diegostoliar** [comment]
In harmony they solve this with a overlay that hides the working area and show only the camera content, its pretty useful.

**Comment by Cat-Butler** [comment]
Possible interim workaround, until this idea is implemented: 
If you tap each layer (working up from the lowest non-Background layer)  in the timeline and "Toggle mask" (i.e. using the Background layer as the mask) it more-or-less works as you posted.
Might not be the best solution if you want to use masking with other timeline layers, and of course it will be tedious if you have lots of timeline layers. Not sure if it will work in all circumstances.

(masked) animation layer
(masked) animation layer
Background layer as mask

Somewhat related idea:
Depending on what you are doing, for camera work you could also create a lower layer rectangle as a mask the same size as the camera view, and make it hierarchy child (hierarchical child?) of the camera (causing it to follow the camera), and mask all other layers to the rectangle, and it will happily mask your layers to only the camera view.
Probably not useful if you need other uses for the hierarchy.

camera
(masked) animation layer
(masked) animation layer
rectangle for camera mask (hidden, hierarchy child of the camera)
infinite Background
